File: backend/src/main/java/com/example/demo/model/Application.java
package com.example.demo.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Application {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "advertisement_id")
	@NotNull
	private Advertisement advertisement;

	@NotBlank
	private String firstName;

	@NotBlank
	private String lastName;

	@NotBlank
	private String email;

	@NotBlank
	private String address;

	@NotBlank
	private String education;

	@NotNull
	private Integer educationLevel;

	@NotNull
	private Double lat;

	@NotNull
	private Double lng;

	@NotBlank
	private String cvLocation;

	@NotBlank
	private String letterLocation;

}

File: frontend/src/app/models/application.ts
import { Advertisement } from "./advertisement";

export interface ApplicationUpload {
    advertisementId: number;
    firstName: string;
    lastName: string;
    email: string;
    address: string;
    education: string;
    educationLevel: number;
    lat: number;
    lng: number;
    cvFile: Blob;
    letterFile: Blob;
}

export interface Application {
    id: number;
    advertisement: Advertisement;
    firstName: string;
    lastName: string;
    email: string;
    address: string;
    education: string;
    educationLevel: number;
    cvLocation: string;
    letterLocation: string;
}
